### ISSUE TITLE
NewDisk code could be optimized ans use the 6809/6309 OpCodes

### ISSUE THREAD

**Issue by chf-6309** [epaell/MECB#1]
Hello, I've a quick look at the code on the MECB_6809/FLEX/newdisk.asm directory, and it seams to be Motorola 6800 assembly code... that could be optimized with 6809 OpCodes, example in the 8 by 8 bits multiplication, that subroutine will be more efficient when use the MUL OpCode, so, AccA * AccB, result on AccD.
Also there are many INX that could be LEAX 1,X, or 2,X, or even use things like LDA sec,X  LDB trk,X... X pointing to the work area, sec and trk are constant (0, 1...) defined on the subroutine.

Anyway great work, for learning ;o)

**Comment by epaell** [comment]
Indeed, there are likely many optimisations that could be made (particularly if the 6309 is considered). I was doing a lot of poking in the dark in terms of getting the FLEX running on the MECB platform and I "think" I just copied these across from an original Horsham Technologies implementation (sadly the original developer passed away and so I wanted to ensure a copy was preserved somewhere - especially as I made a SD-card interface that was closely based on his original design). If I recall correctly, the same code was used for two different systems that he made i.e. a 6800 and 6809, and so likely avoided any specific opcodes that would prevent that from compiling on both systems. In the end, I'm not sure I even used this code as I just simply took an existing working boot disk and removed anything that wasn't needed to make a new "clean" disk :-)
